datajunction-server/alembic/versions/2025_01_23_0229-e865e6d53e6c_indexes_and_unique_constraints_for_.py

```diff
@@ -0,0 +1,35 @@
+"""Indexes and unique constraints for notifiction_preferences table
+
+Revision ID: e865e6d53e6c
+Revises: 0e428c1bb9bc
+Create Date: 2025-01-23 02:29:37.074889+00:00
+
+"""
+# pylint: disable=no-member, invalid-name, missing-function-docstring, unused-import, no-name-in-module
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "e865e6d53e6c"
+down_revision = "0e428c1bb9bc"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table("notification_preferences", schema=None) as batch_op:
+        batch_op.create_index("ix_entity_name", ["entity_name"], unique=False)
```

Comment:
Unless I am missing some of your thinking here...

I don't think we need separate indexes on `entity_name` and `entity_type`, because:
1. the unique_constraint defined on these columns should come with an implicit index on both columns.
2. and because index on `entity_type` will likely be not very effective because of the low cardinality

I would also flip the order of the columns in the unique constraint to `["entity_name", "entity_type"]`, which will effectively serve as an index on the `entity_name` itself.

Reply:
This is great! Thanks Olek, this all makes sense to me. Let me update it.